src/Microphysics/microphysics_diagnostics.jl

```diff
@@ -0,0 +1,194 @@
+    density,
+    saturation_vapor_pressure
+
+import Oceananigans.Grids: prettysummary
+
+struct SaturationSpecificHumidityKernelFunction{μ, FL, M, MF, T, R, TH}
+    flavor :: FL
+    microphysics :: μ
+    microphysical_fields :: M
+    specific_moisture :: MF
+    temperature :: T
+    reference_state :: R
+    thermodynamic_constants :: TH
+end
+
+prettysummary(kf::SaturationSpecificHumidityKernelFunction) = "$(kf.flavor) SaturationSpecificHumidityKernelFunction"
+
+Adapt.adapt_structure(to, k::SaturationSpecificHumidityKernelFunction) =
+    SaturationSpecificHumidityKernelFunction(adapt(to, k.flavor),
+                                             adapt(to, k.microphysics),
+                                             adapt(to, k.microphysical_fields),
+                                             adapt(to, k.specific_moisture),
+                                             adapt(to, k.temperature),
+                                             adapt(to, k.reference_state),
+                                             adapt(to, k.thermodynamic_constants))
+
+const C = Center
+const SaturationSpecificHumidity = KernelFunctionOperation{C, C, C, <:Any, <:Any, <:SaturationSpecificHumidityKernelFunction}
+
+struct Prognostic end
+struct Equilibrium end
+struct TotalMoisture end
+
+"""
+$(TYPEDSIGNATURES)
+
+Return a `KernelFunctionOperation` representing the specified flavor
+of *saturation specific humidity* ``qᵛ⁺`` which correpsonds to `model.microphysics`.
+If `model.microphysics` is not a saturation adjustment scheme, then 
+a warm phase scheme is assumed which computes the saturation specific humidity
+over a planar liquid surface.
+
+## Flavor options
+
+### `:prognostic`
+
+Returns the *saturation specific humidity* corresponding to the `model`'s prognostic state.
+This is the same as the equilibrium saturation specific humidity for saturated conditions
+and a model that uses saturation adjustment microphysics.
+
+### `:equilibrium`
+
+Returns the *saturation specific humidity* in saturated conditions, using the
+`model.specific_moisture`. This is equivalent to the `:total_moisture` flavor
+under saturated conditions with no condensate; or in other words, if `model.specific_moisture` happens
+to be equal to the saturation specific humidity.
+
+### `:total_moisture`
+
+Returns *saturation specific humidity* in the case that the total specific moisture is
+equal to the saturation specific humidity and there is no condensate.
+This is useful for manufacturing perfectly saturated initial conditions.
+
+## Examples
+
+```jldoctestssh
+using Breeze
+grid = RectilinearGrid(size=(1, 1, 128), extent=(1e3, 1e3, 1e3))
+microphysics = SaturationAdjustment()
+model = AtmosphereModel(grid; microphysics)
+set!(model, θ=300)
+qᵛ⁺ = SaturationSpecificHumidity(model, :prognostic)
+
+# output
+KernelFunctionOperation at (Center, Center, Center)
+├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
```

Comment:
How was this output obtained?
```suggestion
├── grid: 1×1×128 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×3 halo
```